miden-lib/asm/sat/internal/note.masm

```diff
@@ -0,0 +1,137 @@
+use.std::crypto::hashes::native
+use.std::mem
+
+use.miden::sat::internal::layout
+
+#! Returns the sender of the note currently being processed. Panics if a note is not being
+#! processed.
+#!
+#! Inputs: []
+#! Outputs: [sender]
+#!
+#! - sender is the sender of the note currently being processed.
+export.get_sender
+    # get the current consumed note pointer
+    exec.layout::get_current_consumed_note_ptr
+    # => [ptr]
+
+    # assert the pointer is not zero - this would suggest the procedure has been called from an
+    # incorrect context
+    dup neq.0 assert
+    # => [ptr]
+
+    # get the sender from the note pointer
+    exec.layout::get_consumed_note_sender
+    # => [sender]
+end
+
+#! Returns the number of assets and vault hash of the note currently being processed. Panics if a
+#! note is not being processed.
+#!
+#! Inputs: []
+#! Outputs: [num_assets, VAULT_HASH]
+#!
+#! - num_assets is the number of assets in the note currently being processed.
+#! - VAULT_HASH is the vault hash of the note currently being processed.
+export.get_vault_data
+    # get the current consumed note pointer
+    exec.layout::get_current_consumed_note_ptr
+    # => [ptr]
+
+    # assert the pointer is not zero - this would suggest the procedure has been called from an
+    # incorrect context
+    dup neq.0 assert
+    # => [ptr]
+
+    # get the number of assets in the note
+    dup exec.layout::get_consumed_note_num_assets
+    # => [num_assets, ptr]
+
+    # get the vault hash from the note pointer
+    swap exec.layout::get_consumed_note_vault_root
+    # => [VAULT_HASH, num_assets]
+end
+
+#! Writes the assets of the currently executing note into memory starting at the specified address.
+#!
+#! Inputs: [dest_ptr]
+#! Outputs: [num_assets, dest_ptr]
+#!
+#! - dest_ptr is the memory address to write the assets.
+#! - num_assets is the number of assets in the currently executing note.
+export.get_assets
+    # TODO: This needs to be changed to a syscall and moved to user facing api
```

Comment:
Is this `TODO` still relevant?

Reply:
see my response here: https://github.com/0xPolygonMiden/miden-base/pull/174#discussion_r1277204465.  I think this is still relevant and has been designed in such a way that allows us to pass the note data across the context boundary.  So yes still relevant until that change is made.